packages/store-ui/src/atoms/Price/Price.tsx

```diff
@@ -0,0 +1,44 @@
+import type { HTMLAttributes, ElementType, ReactNode } from 'react'
+import React, { forwardRef } from 'react'
+
+export type PriceFormatter = (price: number, listing: boolean) => ReactNode
+
+export interface PriceProps
+  extends Omit<HTMLAttributes<HTMLSpanElement>, 'children'> {
+  as?: ElementType
+  testId?: string
+  value: number
+  formatter?: PriceFormatter
+  listing?: boolean
+}
+
+export const Price = forwardRef<Omit<HTMLSpanElement, 'children'>, PriceProps>(
+  function Price(
+    {
+      as: Component = 'span',
+      testId = 'store-price',
+      value,
+      formatter = (price) => price,
+      listing,
+      ...rawProps
+    },
+    ref
+  ) {
+    const formattedPrice = formatter(value, Boolean(listing))
```

Comment:
the `listing` type is boolean, we do not need to use `Boolean` function.
```suggestion
    const formattedPrice = formatter(value, listing)
```

Reply:
It is optional tho. Using `Boolean` ensures that it will be a boolean after this, even if it's undefined.

Reply:
yes, make sense! tks